kolibri/utils/tests/test_main.py

```diff
@@ -71,6 +74,50 @@ def test_version_updated():
     assert not main.should_back_up("0.10.0-dev0", "0.10.0-dev0")
 
 
+@pytest.mark.django_db
+@unittest.skipIf(
+    getattr(settings, "DATABASES")["default"]["ENGINE"] != "django.db.backends.sqlite3",
+    "SQLite only test",
+)
+def test_conditional_backup():
+    import os
+
+    """
+    Tests our db backup logic: conditional_backup, remove all backups
+    """
+    from kolibri.core.deviceadmin.utils import default_backup_folder
+
+    default_path = default_backup_folder()
+    if not os.path.exists(default_path):
```

Comment:
I think the test is still failing because of the backup folder not being clean - in the case when it does exist, you may want to just make sure it is empty here. So force removal, and then just recreate it.